wenet/transducer/search/prefix_beam_search.py

```diff
@@ -0,0 +1,194 @@
+import torch
+from wenet.utils.common import log_add
+class Sequence():
+    def __init__(
+        self,
+        hyp:torch.Tensor,
+        score,
+        h_0:torch.Tensor,
+        c_0: torch.Tensor,
+        last,
+    ):
+        self.hyp = hyp
+        self.score = score
+        self.h_0 = h_0
+        self.c_0 = c_0
+        self.last = last
+
+class PrefixBeamSearch():
+    def __init__(
+        self,
+        encoder,
+        predictor,
+        joint,
+        ctc,
+        sos,
+        blank
+    ):
+        self.encoder = encoder
+        self.predictor = predictor
+        self.joint = joint
+        self.ctc = ctc
+        self.sos = sos
+        self.blank = blank
+
+    def forward_decoder_one_step(
+        self,
+        encoder_x: torch.Tensor,
+        pre_t: torch.Tensor,
+        h_0: torch.Tensor,
+        c_0: torch.Tensor,
+    ):
+        pre_t,h_1, c_1 = self.predictor.forward_step(pre_t.unsqueeze(-1),None, h_0,c_0 )
+        x = self.joint(encoder_x, pre_t)
+        x = x.log_softmax(dim=-1)
+        return x, (h_1,c_1)
+
+    def prefix_beam_search(
+        self,
+        speech: torch.Tensor,
+        speech_lengths: torch.Tensor,
+        decoding_chunk_size: int = -1,
+        beam_size: int = 5,
+        num_decoding_left_chunks: int = -1,
+        simulate_streaming: bool = False,
+        ctc_weight: float= 0.3,
+        transducer_weight: float = 0.7
+    ):
+        """prefix beam search
+           also see wenet.transducer.transducer.beam_search
+        """
+        assert speech.shape[0] == speech_lengths.shape[0]
+        assert decoding_chunk_size != 0
+        device = speech.device
+        batch_size = speech.shape[0]
+        assert batch_size == 1
+
+        # 1. Encoder
+        encoder_out, encoder_mask = self.encoder(
+            speech, speech_lengths, decoding_chunk_size,
+            num_decoding_left_chunks)  # (B, maxlen, encoder_dim)
+        maxlen = encoder_out.size(1)
+        encoder_dim = encoder_out.size(2)
+        running_size = batch_size * beam_size
+        predictor_num = self.predictor.n_layers
+
+        ctc_probs = self.ctc.log_softmax(
+                encoder_out).squeeze(0)
+        beam_init=[]
+
+        # 2. init beam using Sequence to save beam unit
+        h_0,c_0 = self.predictor.init_state(1, method="zero")
```

Comment:
It's  better to add a shape hint for h_0 & c_0

Reply:
we will refine  state to cache in the future, i think there is no need to  hint state shape in the future
plz see: https://github.com/wenet-e2e/wenet/pull/1294/files#diff-5acf73081878302f10e8fd150eb5846e3de4f788f7e6f820c0b3b964af5c6b47

Reply:
Okay